hw2_2/hw2-2/Program.cs

```diff
@@ -0,0 +1,298 @@
+            }
+            else
+            {
+                FloatStackBasedOnList stack = new();
+                float result = calculator.ToCalculate(expression, stack, out correctNotation);
+                if (correctNotation)
+                {
+                    Console.WriteLine($"Result: {result}");
+                    return;
+                }
+
+                Console.WriteLine("Entered expression was incorrect!");
+            }
+        }
+    }
+
+    public class FloatStackBasedOnArray : IStack
+    {
+        private const int defaultArraySize = 20;
+
+        private float[] _ArrayOfOperands;
+
+        private int size = 0;
+
+        public int Size { get; private set; }
+
+        public FloatStackBasedOnArray()
+        {
+            _ArrayOfOperands = new float[defaultArraySize];
+        }
+
+        private float[] ExpandArrayOnDefaultSize(float[] arrayToExpand)
+        {
+            float[] newExpandedArray = new float[arrayToExpand.Length + defaultArraySize];
+            for (int i = 0; i < arrayToExpand.Length; i++)
+            {
+                newExpandedArray[i] = arrayToExpand[i];
+            }
+
+            return newExpandedArray;
+        }
+
+        public void Add(float value)
+        {
+            if (Size == _ArrayOfOperands.Length)
+            {
+                _ArrayOfOperands = ExpandArrayOnDefaultSize(_ArrayOfOperands);
+            }
+
+            _ArrayOfOperands[Size] = value;
+            Size = ++Size;
+        }
+
+        public float Remove(out bool isEmpty)
+        {
+            isEmpty = false;
+            if (Size < 1)
+            {
+                isEmpty = true;
+                return 0;
+            }
+
+            float value = _ArrayOfOperands[Size - 1]; //get head of the stack
+            _ArrayOfOperands[Size - 1] = 0;
+            Size--;
+            return value;
+        }
+    }
+
+    public class FloatStackBasedOnList : IStack
+    {
+        private List<float> _listOfOperands;
+
+        private int size = 0;
+
+        public int Size { get; private set; }
+
+        public FloatStackBasedOnList()
+        {
+            _listOfOperands = new List<float>();
+        }
+
+        public void Add(float value)
+        {
+            _listOfOperands.Add(value);
+            Size = ++Size;
+        }
+
+        public float Remove(out bool isEmpty)
+        {
+            isEmpty = false;
+            if (Size < 1)
+            {
+                isEmpty = true;
+                return 0;
+            }
+
+            float value = _listOfOperands[Size - 1]; //get head of the stack
+            _listOfOperands.RemoveAt(Size - 1);
+            Size--;
+            return value;
+        }
+    }
+
+
+    public interface IStack
+    {
+        void Add(float value);
+
+        float Remove(out bool isEmpty);
+    }
+
+    public class StackCalculatorOfReversePolishNotation
+    {
+        public float delta = 0.0001f; //null comparison precision

```

Comment:
```suggestion
        const float delta = 0.0001f; //null comparison precision
```
Если что-то является константой, нужно делать это константой.
Так и код понятнее, и защита от искушения таки изменить это значение.